src/Castle.Core/DynamicProxy/Contributors/MembersCollector.cs

```diff
@@ -194,6 +194,18 @@ private MetaMethod AddMethod(MethodInfo method, IProxyGenerationHook hook, bool
 		///   to select methods.
 		/// </summary>
 		protected bool AcceptMethod(MethodInfo method, bool onlyVirtuals, IProxyGenerationHook hook)
+		{
+			return AcceptMethodPreScreen(method, onlyVirtuals, hook) && hook.ShouldInterceptMethod(type, method);
+		}
+
+		/// <summary>
+		///   Performs some basic screening to filter out non-interceptable methods.
+		/// </summary>
+		/// <remarks>
+		///   The <paramref name="hook"/> will get invoked for non-interceptable method notification only;
+		///   it does not get asked whether or not to intercept the <paramref name="method"/>.
+		/// </remarks>
+		protected bool AcceptMethodPreScreen(MethodInfo method, bool onlyVirtuals, IProxyGenerationHook hook)
```

Comment:
I am splitting off the pre-screening from `AcceptMethod` so that `WrappedMembersCollector` has a way of distinguishing between non-interceptable methods (e.g. `object.Finalize`, `object.MemberwiseClone`), and methods that the hook wants filtered out. It is only the latter group of methods that should be forwarded to the target; the former group of methods must not be proxied at all!

(In my original PR, I made this distinction via `AllMethodsHook.SkippedTypes`, which wasn't ideal; see https://github.com/castleproject/Core/pull/450#discussion_r303399484.)